email-service/internal/broker/subscriber/client.go

```diff
@@ -0,0 +1,86 @@
+package subscriber
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"time"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport/config"
+)
+
+var subTimeout = 5 * time.Second
+
+type Handler func(ctx context.Context, email string) error
+
+const (
+	subscribedEventType   = "Subscribe"
+	unsubscribedEventType = "Unsubscribe"
+)
+
+type SubscribeEvent struct {
```

Comment:
I don't quite get the point of this structure.

If this is intended to be just a generic wrapper around rabbit mq, kind of facade - OK, but why are you trying to give this struct more specific meaning, binding it to the context of a specific use case (add/delete subscribers)?

What I mean - is if this is just a facade/wrapper around rabbit mq lib I think it'd be better to have only one method `Subscribe(topic string, fn Handler) error`, this way this wrapper won't know why, and where it's used. And this way wrapper can be reused across different use cases.

On the opposite side, if this is not a wrapper/facade, but a Transport layer (kind of handler but not for HTTP but for AMQP) specifically for subscribing unsubscribing users, then why are you accepting a function handler as an argument to the method rather than receiving handler through DI when this struct is created?

Reply:
This was designed more like a transport layer (handler, if you wish) specifically for sub/unsub, but also to have multiple handlers listening for the same sub/unsub event. For now, the functionality of _multiple_ listeners is not required (which is indeed overengineering), but isn't that okay?

Reply:
ok, let it be